pkgs/applications/gis/tunnelx/default.nix

```
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, jdk
+, jre
+, fetchgit
+, survex
+, makeWrapper
+, bash
+, coreutils
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tunnelx";
+  version = "v2023-07-nix";
+
+  src = fetchFromGitHub {
+    owner = "CaveSurveying";
+    repo = "tunnelx";
+    rev = "8c455ec27e4e9ea754cd87a4e2e0f0b2bc558b88";
+    hash = "sha256-r70ZmrpN7I/bsvHAxE7uRaryqNlz/z9zA55qiCC8NkE=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    jdk
+  ];
+
+  runtimeInputs = [
+    survex
+  ];
+
+  buildPhase = ''
+    javac -d . src/*.java
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/java
+    cp -r Tunnel $out/java
+    cp -r symbols $out/java
+    cp -r tutorials $out/java
+    makeWrapper ${jre}/bin/java $out/bin/tunnelx \
+        --add-flags "-cp $out/java Tunnel.MainBox" \
+        --set SURVEX_EXECUTABLE_DIR ${survex}/bin/ \
+        --set TUNNEL_USER_DIR $out/java/
+  '';
+
+  meta = with lib; {
+    description = "TunnelX – cave surveying software";
```

```suggestion
    description = "A cave surveying software";
```
we don't have the pname at the beginning of a description in nixpkgs